backend/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Controller,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  Inject,
+  Request,
+  Get,
+  HttpStatus,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { AuthService } from './auth.service';
+import { RequestWithRoles, RolesRequest } from './interface';
+import { AppLogger } from '../logger/logger.service';
+
+@Controller('auth')
+@ApiTags('Auth API')
+@UseInterceptors(ClassSerializerInterceptor)
+export class AuthController {
+  constructor(
+    @Inject(AuthService) private readonly authService: AuthService,
+    private readonly logger: AppLogger,
+  ) {
+    this.logger.setContext(AuthController.name);
+  }
+
+  @ApiOperation({
+    summary: 'Verify the role and sent user data',
+    description: 'Returns the user data with the request',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+  })
+  @Get()
+  async checkUserPermissions(
+    @Request() req: RequestWithRoles,
+  ): Promise<RolesRequest> {
+    this.logger.log(
+      `${req.method}: ${req.url} - ${req.role} - ${req.username}`,
+    );
+
+    const { isMember, isSupervisor } =
+      await this.authService.verifyMemberOrSupervisor(req.idir);
+
+    return {
+      role: req?.role,
+      program: req?.program,
+      username: `${req?.username}`,
+      idir: req?.idir,
```

Comment:
At least with local, username is returning the full name and idir is returning the email. Worth a check if this will be the case in SSO.